pkg/cluster/sync.go

```diff
@@ -229,6 +229,7 @@ func (c *Cluster) syncStatefulSet() error {
 	var (
 		podsRollingUpdateRequired bool
 	)
+	// the following code should not return before the end of this function in a non-error case
```

Comment:
1. "The following" == code until the end of the function?
2. Can the code leave the cluster in inconsistent state in the case of early non-error return ?

Reply:
The rolling update is applied at the end of the function, and a code that will return beforehand will miss the rolling update.